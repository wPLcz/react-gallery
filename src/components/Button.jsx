import React, { useEffect } from 'react';
import styled  from 'styled-components';
import PropTypes from 'prop-types';

const Button = (props) => {
  return (
    <StyledButton active={props.active} onClick={() => props.takeNewestSorting(props.id)}>
      {props.children}
    </StyledButton>)
};

const StyledButton = styled.button`
    padding: 0;
    margin: 0 15px;
    border: none;
    background-color: transparent;
    position: relative;
    text-align: center;
    cursor: pointer;
    color: ${({ active }) =>  active  ? '#CB626E;' : '#999'};
    font-size: 14px;
    font-family: Helvetica;
    text-transform: uppercase;
    
    &::after {
    position: absolute;
    display: block;
    content: '';
    bottom: -10px;
    background-color: ${({ active }) =>  active ? '#CB626E;' : '#999'};
    width: 0%;
    height: 2px;
    transition: width 250ms ease-in-out;
    }

    &:hover {
      &::after {
        width: 100%;
      }   
    }
    
    @media (min-width: 960px) {
      margin: 0 25px;
      font-size: 24px;
    }
`;

Button.propTypes = {
  active: PropTypes.bool,
  id: PropTypes.string,
  takeNewestSorting: PropTypes.func,
};

Button.defaultProps = {
  active: false,
  id: '',
  takeNewestSorting: ()=> {},
};

export default Button;
