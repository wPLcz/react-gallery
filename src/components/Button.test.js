import React from "react";
import ReactDOM from "react-dom";
import renderer from 'react-test-renderer';
import Button from "./Button";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Button/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly when there are no items', () => {
  const tree = renderer.create(<Button/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly ', () => {
  const tree = renderer.create(<Button/>).toJSON();
  expect(tree).toMatchSnapshot();
});
