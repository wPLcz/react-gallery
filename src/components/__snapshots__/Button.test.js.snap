// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly  1`] = `
<button
  className="sc-bdnylx etDzgm"
  onClick={[Function]}
/>
`;

exports[`renders correctly when there are no items 1`] = `
<button
  className="sc-bdnylx etDzgm"
  onClick={[Function]}
/>
`;
